--- ../src-base/minecraft/net/minecraft/network/PacketThreadUtil.java
+++ ../src-work/minecraft/net/minecraft/network/PacketThreadUtil.java
@@ -8,12 +8,8 @@
     {
         if (!p_180031_2_.func_152345_ab())
         {
-            p_180031_2_.func_152344_a(new Runnable()
-            {
-                public void run()
-                {
-                    p_180031_0_.func_148833_a(p_180031_1_);
-                }
+            p_180031_2_.func_152344_a(() -> {
+                p_180031_0_.func_148833_a(p_180031_1_);
\ No newline at end of file
             });
             throw ThreadQuickExitException.field_179886_a;
         }
