--- ../src-base/minecraft/net/minecraft/block/BlockMycelium.java
+++ ../src-work/minecraft/net/minecraft/block/BlockMycelium.java
@@ -37,30 +37,6 @@
 
     public void func_180650_b(World p_180650_1_, BlockPos p_180650_2_, IBlockState p_180650_3_, Random p_180650_4_)
     {
-        if (!p_180650_1_.field_72995_K)
-        {
-            if (p_180650_1_.func_175671_l(p_180650_2_.func_177984_a()) < 4 && p_180650_1_.func_180495_p(p_180650_2_.func_177984_a()).func_185891_c() > 2)
-            {
-                p_180650_1_.func_175656_a(p_180650_2_, Blocks.field_150346_d.func_176223_P().func_177226_a(BlockDirt.field_176386_a, BlockDirt.DirtType.DIRT));
-            }
-            else
-            {
-                if (p_180650_1_.func_175671_l(p_180650_2_.func_177984_a()) >= 9)
-                {
-                    for (int i = 0; i < 4; ++i)
-                    {
-                        BlockPos blockpos = p_180650_2_.func_177982_a(p_180650_4_.nextInt(3) - 1, p_180650_4_.nextInt(5) - 3, p_180650_4_.nextInt(3) - 1);
-                        IBlockState iblockstate = p_180650_1_.func_180495_p(blockpos);
-                        IBlockState iblockstate1 = p_180650_1_.func_180495_p(blockpos.func_177984_a());
-
-                        if (iblockstate.func_177230_c() == Blocks.field_150346_d && iblockstate.func_177229_b(BlockDirt.field_176386_a) == BlockDirt.DirtType.DIRT && p_180650_1_.func_175671_l(blockpos.func_177984_a()) >= 4 && iblockstate1.func_185891_c() <= 2)
-                        {
-                            p_180650_1_.func_175656_a(blockpos, this.func_176223_P());
-                        }
-                    }
-                }
-            }
-        }
     }
 
     @SideOnly(Side.CLIENT)
