--- ../src-base/minecraft/net/minecraft/block/BlockFarmland.java
+++ ../src-work/minecraft/net/minecraft/block/BlockFarmland.java
@@ -8,8 +8,6 @@
 import net.minecraft.block.state.BlockStateContainer;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.Entity;
-import net.minecraft.entity.EntityLivingBase;
-import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.init.Blocks;
 import net.minecraft.item.Item;
 import net.minecraft.util.EnumFacing;
@@ -72,30 +70,17 @@
 
     public void func_180658_a(World p_180658_1_, BlockPos p_180658_2_, Entity p_180658_3_, float p_180658_4_)
     {
-        if (!p_180658_1_.field_72995_K && p_180658_1_.field_73012_v.nextFloat() < p_180658_4_ - 0.5F && p_180658_3_ instanceof EntityLivingBase && (p_180658_3_ instanceof EntityPlayer || p_180658_1_.func_82736_K().func_82766_b("mobGriefing")) && p_180658_3_.field_70130_N * p_180658_3_.field_70130_N * p_180658_3_.field_70131_O > 0.512F)
-        {
-            func_190970_b(p_180658_1_, p_180658_2_);
-        }
-
         super.func_180658_a(p_180658_1_, p_180658_2_, p_180658_3_, p_180658_4_);
     }
 
     protected static void func_190970_b(World p_190970_0_, BlockPos p_190970_1_)
     {
-        p_190970_0_.func_175656_a(p_190970_1_, Blocks.field_150346_d.func_176223_P());
-        AxisAlignedBB axisalignedbb = field_194405_c.func_186670_a(p_190970_1_);
-
-        for (Entity entity : p_190970_0_.func_72839_b((Entity)null, axisalignedbb))
-        {
-            double d0 = Math.min(axisalignedbb.field_72337_e - axisalignedbb.field_72338_b, axisalignedbb.field_72337_e - entity.func_174813_aQ().field_72338_b);
-            entity.func_70634_a(entity.field_70165_t, entity.field_70163_u + d0 + 0.001D, entity.field_70161_v);
-        }
     }
 
     private boolean func_176529_d(World p_176529_1_, BlockPos p_176529_2_)
     {
         Block block = p_176529_1_.func_180495_p(p_176529_2_.func_177984_a()).func_177230_c();
-        return block instanceof BlockCrops || block instanceof BlockStem;
+        return block instanceof net.minecraftforge.common.IPlantable && canSustainPlant(p_176529_1_.func_180495_p(p_176529_2_), p_176529_1_, p_176529_2_, EnumFacing.UP, (net.minecraftforge.common.IPlantable)block);
     }
 
     private boolean func_176530_e(World p_176530_1_, BlockPos p_176530_2_)
@@ -108,7 +93,7 @@
             }
         }
 
-        return false;
+        return net.minecraftforge.common.FarmlandWaterManager.hasBlockWaterTicket(p_176530_1_, p_176530_2_);
\ No newline at end of file
     }
 
     public void func_189540_a(IBlockState p_189540_1_, World p_189540_2_, BlockPos p_189540_3_, Block p_189540_4_, BlockPos p_189540_5_)
